polaris-core/src/main/java/org/apache/polaris/core/config/BehaviorChangeConfigurationSince.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.config;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that should be applied to all BehaviorChangeConfiguration instances. The value here
```

Comment:
Should we add validation for that to the spotless check too?

Reply:
Definitely. I looked into this a bit, but it seems like the "best" way would be via reflection (enumerating all instances of the BehaviorChangeConfiguration type) which feels hacky. At the very least, let me add a check so that no configs are created outside of the correct file.

Reply:
That can be done in a compile-time annotation processor (I think it has access to all elements, even if not annotated).

Reply:
Awesome, I'll look into that and update the PR. For now, I added a lint check to try and make sure the flags are contained within the one file.

Reply:
After some research, I have one doubt about the above approach. If we want to scan for BehaviorChangeConfiguration instances _without_ the annotation, is an annotation processor really the right choice? I am wondering if this, too, needs to be done in the linter.

Reply:
Cross references are inevitable. For example, immutables generate code from annotations that is referenced by some other hand-written classes. `@Overrides` annotations relate to base classes, which do not have the annotation itself. I think it is quite normal for annotation processors to "look around" in the codebase.

By the way, the immutables lib can be a source of inspiration here, I guess :wink:

Reply:
I tried adding a processor `BehaviorChangeProcessor` and registering it, but I don't think it actually runs based on my testing. If you're able to take a look and you see a problem, please do let me know